src/devtools/client/webconsole/webconsole.js

```diff
@@ -229,25 +229,26 @@ class WebConsole {
     return this._highlighter;
   }
 
-  async openNodeInInspector(grip) {
+  async openNodeInInspector(valueFront) {
     if (!this.toolbox) {
       return;
     }
 
     const onSelectInspector = this.toolbox.selectTool("inspector", "inspect_dom");
 
-    const onNodeFront = this.toolbox.target
-      .getFront("inspector")
-      .then(inspectorFront => inspectorFront.getNodeFrontFromNodeGrip(grip));
+    const onNodeFront = valueFront
+      .getPause()
+      .ensureDOMFrontAndParents(valueFront._object.objectId)
+      .then(async nf => {
+        await nf.ensureParentsLoaded();
+        return nf;
+      });
```

Comment:
This isn't just a style thing: my code returns `Promise<NodeFront>` but yours returns `Promise<undefined[]>`.

And yes, this is independent of all the other stuff in this PR (and also applies to the old markup view), just tell me if you want me to move it into a separate PR.

Reply:
@hbenl i think mine does as well :) 

I think this would be better in its own PR for tracking purposes. The markup view is big so itd be easy to get confused in the future